RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticSimClusterMapper.h

```diff
@@ -0,0 +1,53 @@
+#ifndef __RecoParticleFlow_PFClusterProducer_RealisticSimClusterMapper_H__
+#define __RecoParticleFlow_PFClusterProducer_RealisticSimClusterMapper_H__
+/////////////////////////
+// Author: Felice Pantaleo
+// Date:   30/06/2017
+// Email: felice@cern.ch
+/////////////////////////
+#include "RecoParticleFlow/PFClusterProducer/interface/InitialClusteringStepBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFraction.h"
+#include "RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h"
+
+#include "SimDataFormats/CaloAnalysis/interface/SimClusterFwd.h"
+
+class RealisticSimClusterMapper : public InitialClusteringStepBase {
+ public:
+ RealisticSimClusterMapper(const edm::ParameterSet& conf,
+			 edm::ConsumesCollector& sumes) :
+    InitialClusteringStepBase(conf,sumes),
+    _invisibleFraction(conf.getParameter<double>("invisibleFraction")),
+    _exclusiveFraction(conf.getParameter<double>("exclusiveFraction")),
+    _useMCFractionsForExclEnergy(conf.getParameter<bool>("useMCFractionsForExclEnergy"))
+    {
+      _simClusterToken = sumes.consumes<SimClusterCollection>(conf.getParameter<edm::InputTag>("simClusterSrc"));
+
+
+    }
+  virtual ~RealisticSimClusterMapper() {}
+  RealisticSimClusterMapper(const RealisticSimClusterMapper&) = delete;
+  RealisticSimClusterMapper& operator=(const RealisticSimClusterMapper&) = delete;
+
+  virtual void updateEvent(const edm::Event&) override final;
+  virtual void update(const edm::EventSetup&) override final;
+
+  void buildClusters(const edm::Handle<reco::PFRecHitCollection>&,
+		     const std::vector<bool>&,
+		     const std::vector<bool>&, 
+		     reco::PFClusterCollection&) override;
+  
+ private:  
+  hgcal::RecHitTools _rhtools;
```

Comment:
ciao Andrea, 
I just copy-pasted this [class:](https://github.com/felicepantaleo/cmssw/blob/93214d33a0bc596b4ea88cc052b3061d6e9a8e24/RecoParticleFlow/PFClusterProducer/plugins/SimMappers/GenericSimClusterMapper.h#L29)

I will anyway update as you ask.